packages/dropdownList/src/Default/Default.tsx

```diff
@@ -0,0 +1,210 @@
+          {
+            fields: ['theme'],
+            value: 'Theme',
+          },
+          {
+            fields: ['theme', 'appearance'],
+            value: 'Theme',
+          },
+          {
+            fields: ['size'],
+            value: 'Size',
+          },
+        ],
+      },
+    },
+    ['prefix', 'theme', 'appearance', 'size']
+  );
+
+  const useKeyPress = (targetKey: string) => {
+    const [keyPressed, setKeyPressed] = React.useState(false);
+
+    const downHandler = ({ key }: { key: string }) => {
+      if (key === targetKey) {
+        setKeyPressed(true);
+      }
+    };
+
+    const upHandler = ({ key }: { key: string }) => {
+      if (key === targetKey) {
+        setKeyPressed(false);
+      }
+    };
+
+    React.useEffect(() => {
+      window.addEventListener('keydown', downHandler);
+      window.addEventListener('keyup', upHandler);
+
+      return () => {
+        window.removeEventListener('keydown', downHandler);
+        window.removeEventListener('keyup', upHandler);
+      };
+    });
+    return keyPressed;
+  };
+
+  const downPress = useKeyPress('ArrowDown');
+  const upPress = useKeyPress('ArrowUp');
+  const enterPress = useKeyPress('Enter');
+  const tabPress = useKeyPress('Tab');
+
+  React.useEffect(() => {
+    if (dropdown?.items.length && downPress) {
+      if (!dropdownOpen) {
+        setDropdownOpen(true);
+      }
+      if (dropdown.items.length) {
+        setCursor(prevState =>
+          typeof dropdown?.items?.length === 'number' &&
+          prevState < dropdown?.items?.length - 1
+            ? prevState + 1
+            : prevState
+        );
+      }
+    }
+    setSelected(
+      dropdown?.items[cursor as unknown as keyof DropDownContentItemProps]
+    );
+    const selectedItem = document.querySelector(
+      '.owlui-dropdown-menu-selectedMenuItem'
+    );
+    selectedItem?.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+  }, [downPress]);
+  React.useEffect(() => {
+    if (dropdown?.items.length && upPress) {
+      if (!dropdownOpen) {
+        setDropdownOpen(true);
+      }
+      setCursor(prevState => (prevState > 0 ? prevState - 1 : prevState));
+    }
+    setSelected(
+      dropdown?.items[cursor as unknown as keyof DropDownContentItemProps]
+    );
+    const selectedItem = document.querySelector(
+      '.owlui-dropdown-menu-selectedMenuItem'
+    );
+    selectedItem?.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+  }, [upPress]);
+  React.useEffect(() => {
+    if (dropdown?.items.length && enterPress) {
+      setSelected(
+        dropdown?.items[cursor as unknown as keyof DropDownContentItemProps]
+      );
+    }
+  }, [cursor, enterPress]);
+  React.useEffect(() => {
+    if (dropdown?.items.length && tabPress) {
+      if (dropdownOpen) {
+        setDropdownOpen(false);
+      }
+    }
+  }, [tabPress]);
+  const handleClickOutside = (ref: React.MutableRefObject<HTMLDivElement>) => {
+    React.useEffect(() => {
```

Comment:
We need to find an alternative to `useEffect` here. `useEffect` at least once after every render, if you add a console.log to each one you will see they run a lot. 

I suggest you try [this approach](https://towardsdev.com/how-to-create-an-accessible-custom-select-dropdown-in-react-9d9858415f10) to handle the keyboard navigation and the display changes.

You will notice we don't need to add any event listener to the window, instead, we define a function directly to the element, and we don't actually need `useEffect`, we can handle everything only by using the state.